apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,268 @@
+{
+	int ret = 0;
+	int total_write = 0;
+
+	long t = get_time();
+	while (total_write < nbytes) {
+		ret = write(fd, buf + total_write, nbytes - total_write);
+		if (ret == 0 || ret == -1) {
+			close(fd);
+			exit(EXIT_FAILURE);
+		}
+		total_write += ret;
+	}
+	long dt = get_time() - t;
+
+	g_total_time += dt;
+	rw_testlog("write:%dms, ", dt);
+}
+
+static void _rwt_read(int fd, char *buf, size_t nbytes)
+{
+	int ret = 0;
+	int total_read = 0;
+
+	long t = get_time();
+	while (total_read < nbytes) {
+		ret = read(fd, buf + total_read, nbytes - total_read);
+		if (ret == 0 || ret == -1) {
+			close(fd);
+			exit(EXIT_FAILURE);
+		}
+		total_read += ret;
+	}
+	long dt = get_time() - t;
+
+	g_total_time += dt;
+	rw_testlog("read:%dms, ", dt);
+}
+
+static void wr_test(void *buf, size_t nbytes)
+{
+	int i = 0;
+	int fd = -1;
+	for (; i < g_iteration; i++) {
+		fd = _rwt_open('w');
+		_rwt_write(fd, buf, nbytes);
+		_rwt_close(fd);
+
+		fd = _rwt_open('r');
+		_rwt_read(fd, buf, nbytes);
+		_rwt_close(fd);
+	}
+}
+
+static void w_test(void *buf, size_t nbytes)
+{
+	int i = 0;
+	int fd = -1;
+	for (; i < g_iteration; i++) {
+		fd = _rwt_open('w');
+		_rwt_write(fd, buf, nbytes);
+		_rwt_close(fd);
+	}
+}
+
+static void r_test(char *buf, size_t nbytes)
+{
+	int fd = -1;
+	int total_write = 0;
+	int ret = 0;
+	fd = open(g_filepath, O_WRONLY | O_CREAT | O_TRUNC);
+	if (fd == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	while (total_write < nbytes) {
+		ret = write(fd, buf + total_write, nbytes - total_write);
+		if (ret == 0 || ret == -1) {
+			close(fd);
+			printf("Write Fail In %s(%d)\n", __FUNCTION__, errno);
+			exit(EXIT_FAILURE);
+		}
+		total_write += ret;
+	}
+	close(fd);
+
+	int i = 0;
+	for (; i < g_iteration; i++) {
+		fd = _rwt_open('r');
+		_rwt_read(fd, buf, nbytes);
+		_rwt_close(fd);
+	}
+}
+
+void f_rw_performance(int size, int operation)
+{
+	char buf[size];
+
+	switch (operation) {
+	case 1:
+		wr_test(buf, sizeof(buf));
+		break;
+	case 2:
+		r_test(buf, sizeof(buf));
+		break;
+	case 3:
+		w_test(buf, sizeof(buf));
+		break;
+	default:
+		return;
+	}
+	printf("run_time: %d ms\n", g_total_time);
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+	int fs_rw_test_main(int argc, char *argv[])
+#endif
+{
+	int size = 0;
+	int operation = 0;
+
+	printf("FS R/W Performance Test!\n");
+	g_total_time = 0;
+	if (argc <= 5) {
+		printf("%s", USAGE);
+		return 0;
+	}
+
+	size = atoi(argv[2]);
+	g_iteration = atoi(argv[3]);
+	operation = atoi(argv[4]);
+
+	if (operation < 0 || operation > 3 ||
+		size > FS_RW_SIZE_MAX || size < 0 ||
+		g_iteration < 1 || g_iteration > FS_RW_ITERATION_MAX) {
+		printf("%s", USAGE);
+		return -1;
+	}
+
+	if (strlen(argv[5]) != 1 ||
+		(argv[5][0] != 'y' && argv[5][0] != 'n')) {
+		printf("%s", USAGE);
+		return -1;
+	}
+	g_printopt = argv[5][0];
+	g_filepath = argv[1];
+
+	f_rw_performance(size, operation);
```

Comment:
Could you let me know the reason of delivering only size and operation among 5 arguments? why not print and iteration?